src/Controller/ImportController.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * BEdita, API-first content management framework
+ * Copyright 2018 ChannelWeb Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+namespace App\Controller;
+
+use BEdita\SDK\BEditaClientException;
+use Cake\Core\Configure;
+use Cake\Event\Event;
+use Cake\Http\Response;
+
+/**
+ * Import controller: upload and load using filters
+ */
+class ImportController extends AppController
+{
+    /**
+     * Display import page.
+     *
+     * @return void
+     */
+    public function index() : void
+    {
+        $this->loadFilters();
+    }
+
+    /**
+     * Import data by filter/file.
+     *
+     * @return \Cake\Http\Response|null the Response.
+     */
+    public function file() : ?Response
+    {
+        try {
+            // prepare import filter
+
+            $filter = $this->request->getData('filter');
+            if (empty($filter)) {
+                throw new BEditaClientException('Import filter not selected', 500);
```

Comment:
It is not a BEdita API client related exception, I'd use a simple `BadRequestException` - without a `catch` it should simply not happen